R/dataframe.R

```diff
@@ -157,6 +160,9 @@ as_data_frame.data.frame <- function(x, ...) {
 #' @export
 #' @rdname as_data_frame
 as_data_frame.list <- function(x, validate = TRUE, ...) {
+
+  x <- compact(x)
```

Comment:
Crashing dplyr is neither here nor there, because you could argue that's a dplyr bug that should be fixed.

I think you'd be better off taking the position that a data frame can only contain vectors and NULL is not a vector hence is not allowed. There be existing check for non-vectors (e.g. environments, symbols etc) that throws an informative error, rather than special casing NULL here.  

But beware `is.atomic(NULL)` == `TRUE` so you might want to bring in purrr's `is_atomic`:

```R
is_atomic <- function(x) function(x) {
  typeof(x) %in% c("logical", "integer", "double", "complex", "character", "raw")
}
```